vscode-dotnet-runtime-library/src/Acquisition/DotnetPathFinder.ts

```diff
@@ -144,10 +162,10 @@ Bin Bash Path: ${os.platform() !== 'win32' ? (await this.executor?.execute(Comma
 
         options.dotnetInstallToolCacheTtlMs = DOTNET_INFORMATION_CACHE_DURATION_MS;
         const findCommand = CommandExecutor.makeCommand(pathLocatorCommand, ['dotnet']);
-        const dotnetsOnPATH = (await this.executor?.execute(findCommand, options))?.stdout.split('\n').map(x => x.trim()).filter(x => x !== '');
+        const dotnetsOnPATH = (await this.executor?.execute(findCommand, options, false))?.stdout.split('\n').map(x => x.trim()).filter(x => x !== '');

```

Comment:
is this looking at the PATH environment variable to see if there's anything called .NET or is it trying to resolve the dotnet executable for each directory in the path variable to see whether it exists?

Reply:
Its using which or where to list every possible thing called 'dotnet' and later it will check if the file works and is the right version + arch of dotnet, etc etc